packages/@aws-cdk/aws-secretsmanager/test/test.secret.ts

```diff
@@ -435,6 +435,215 @@ export = {
     test.done();
   },
 
+  'grantWrite'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const secret = new secretsmanager.Secret(stack, 'Secret', {});
+    const role = new iam.Role(stack, 'Role', { assumedBy: new iam.AccountRootPrincipal() });
+
+    // WHEN
+    secret.grantWrite(role);
+
+    // THEN
+    expect(stack).to(haveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [{
+          Action: 'secretsmanager:PutSecretValue',
+          Effect: 'Allow',
+          Resource: { Ref: 'SecretA720EF05' },
+        }],
+      },
+    }));
+    test.done();
+  },
+
+  'grantWrite with kms'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const key = new kms.Key(stack, 'KMS');
+    const secret = new secretsmanager.Secret(stack, 'Secret', { encryptionKey: key });
+    const role = new iam.Role(stack, 'Role', { assumedBy: new iam.AccountRootPrincipal() });
+
+    // WHEN
+    secret.grantWrite(role);
+
+    // THEN
+    const expectStack = expect(stack);
+    expectStack.to(haveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [{
+          Action: 'secretsmanager:PutSecretValue',
+          Effect: 'Allow',
+          Resource: { Ref: 'SecretA720EF05' },
+        }],
+      },
+    }));
+    expectStack.to(haveResource('AWS::KMS::Key', {
```

Comment:
Can we use `haveResourceLike` instead, to match only part of the resource definition here? (Only the one that's relevant to this test)

Reply:
Turns out the answer is 'no, we can't, unless there's some trick that I don't know for arrays of IAM statements.'

The implementation for `haveResourceLike` will always reject arrays of different length ( https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/assert/lib/assertions/have-resource.ts#L140 ), and the differing parts here are the list/array of IAM policy statements.

Fixing/enhancing `HaveResourceAssertion` to be able to figure out subsets of IAM policy statements would be super useful, but definitely out-of-scope for this PR; it's also likely to have larger ramifications that would require some thought.

It's a shame too, there's a tonne of duplication of the KMS policy in the secrets tests that would be nice to de-duplicate.

Reply:
That is true, it's not ideal, but it allows you to write:

```ts
    expectStack.to(haveResourceLike('AWS::KMS::Key', {
      KeyPolicy: {
        Statement: [
          {},
          {},
          {},
          {
            Action: [
              'kms:Encrypt',
              'kms:ReEncrypt*',
              'kms:GenerateDataKey*',
            ],
            Condition: {
              StringEquals: {
                'kms:ViaService': {
                  'Fn::Join': ['', [
                    'secretsmanager.',
                    { Ref: 'AWS::Region' },
                    '.amazonaws.com',
                  ]],
                },
              },
            },
            Principal: {
              AWS: {
                'Fn::GetAtt': [
                  'Role1ABCC5F0',
                  'Arn',
                ],
              },
            },
            Resource: '*',
          },
        ],
      },
    });
```

Which is quite a bit more readable.

Reply:
Ahhh. So, there *is* a trick. :-)

I'll apply that, and get a new version up later today. I'll probably also clean up the other tests in the file while I'm at it...

Reply:
@skinny85 Done. I cleaned up the tests for `grantRead` to use `haveResourceLike` as well.